### ISSUE TITLE
[assume-role-dev] in credentials file cause plugin exception

### ISSUE THREAD

**Issue by vesdakon** [aws/aws-toolkit-jetbrains#777]
**Describe the bug**
Adding [assume-role-dev] in credentials file causes Exception and PyCharm failes to restart. Credentials file:
[default]
aws_access_key_id=***
aws_secret_access_key=**
region = eu-west-1 

[assume-role-dev]
role_arn = ****
source_profile = default
output = json
region = eu-west-1 

Log file:
2019-02-01 12:02:03,833 [   9046]  ERROR - nSystem.impl.ActionManagerImpl - PyCharm 2018.3.3  Build #PY-183.5153.39 
2019-02-01 12:02:03,833 [   9046]  ERROR - nSystem.impl.ActionManagerImpl - JDK: 1.8.0_152-release; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o 
2019-02-01 12:02:03,834 [   9047]  ERROR - nSystem.impl.ActionManagerImpl - OS: Windows 10 
2019-02-01 12:02:03,834 [   9047]  ERROR - nSystem.impl.ActionManagerImpl - Plugin to blame: AWS Toolkit version: 1.2-EAP.2019.01.25 
2019-02-01 12:02:03,991 [   9204]  ERROR - nSystem.impl.ActionManagerImpl - software.aws.toolkits.jetbrains.components.telemetry.AnActionWrapper [Plugin: aws.toolkit] 
com.intellij.diagnostic.PluginException: software.aws.toolkits.jetbrains.components.telemetry.AnActionWrapper [Plugin: aws.toolkit]
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:149)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getActionImpl(ActionManagerImpl.java:471)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getAction(ActionManagerImpl.java:459)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.preloadActions(ActionManagerImpl.java:1275)
	at com.intellij.openapi.actionSystem.impl.ActionPreloader.preload(ActionPreloader.java:32)
	at com.intellij.openapi.application.Preloader.lambda$null$0(Preloader.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.application.Preloader.lambda$initComponent$1(Preloader.java:63)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:227)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:26)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2$1.run(BoundedTaskExecutor.java:200)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:229)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:194)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: software.aws.toolkits.jetbrains.components.telemetry.AnActionWrapper
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:145)
	... 19 more

**To reproduce**
In PyCharm install latest the latest AWS Toolkit plugin and restart IDE. Restarting fails with exception notification. Removing [assume-role-dev] from credentials file does not cause failure. My company uses role-based login, so assume-role-dev is needed. SAM CLI works as expected with the above credential file.

**Expected behavior**
Should be able to explore resources in AWS

**Your Environment**
 - OS:  Windows 10
 - JetBrains' Product: Pycharm 
 - JetBrains' Product Version: 2018.3.3
 - Toolkit Version: AWS Toolkit v1.2-EAP.2019.01.25
 - SAM CLI Version: SAM CLI, version 0.10.0
 - JVM/Python Version: Python 3.6.6

**Comment by abrooksv** [comment]
The true exception that led to the NoClassDefFound will be higher in the idea.log

**Comment by vesdakon** [comment]
The complete idea.log is as follows:
[idea.log](https://github.com/aws/aws-toolkit-jetbrains/files/2889094/idea.log)

**Comment by abrooksv** [comment]
Thanks for getting back to us.

```
Caused by: java.lang.IllegalStateException: Invalid profile file: Circular relationship detected with profiles [default].
	at software.amazon.awssdk.utils.Validate.validState(Validate.java:384)
	at software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.roleBasedProfileCredentialsProvider(ProfileCredentialsUtils.java:137)
	at software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.credentialsProvider(ProfileCredentialsUtils.java:90)
	at software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.lambda$roleBasedProfileCredentialsProvider$0(ProfileCredentialsUtils.java:146)
	at java.util.Optional.flatMap(Optional.java:241)
	at software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.roleBasedProfileCredentialsProvider(ProfileCredentialsUtils.java:145)
	at software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.credentialsProvider(ProfileCredentialsUtils.java:90)
	at software.amazon.awssdk.auth.credentials.internal.ProfileCredentialsUtils.credentialsProvider(ProfileCredentialsUtils.java:80)
	at software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider.lambda$new$0(ProfileCredentialsProvider.java:65)
	at java.util.Optional.flatMap(Optional.java:241)
```

An issue in the Java SDK always causes the default credential chain to load, this causes the profiles to load (and it sees your default profile as invalid which kills the toolkit due to the exception, I assume you have `source_profile: default` in it that is triggering it.

The java SDK team fixed the initialization issue, our EAP builds should load for you so that the toolkit will start.

The fact that it sees your default profile as invalid though will mean that profile can't be used within the toolkit though.

**Comment by abrooksv** [comment]
Fix has been released in 1.2
